x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtDateClaimValidatorTests.java

```diff
@@ -0,0 +1,129 @@
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
+
+import java.text.ParseException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class JwtDateClaimValidatorTests extends ESTestCase {
+    private Clock clock;
+
+    @Before
+    public void init() {
+        clock = mock(Clock.class);
+    }
+
+    public void testClaimIsNotDate() throws ParseException {
+        final String claimName = randomAlphaOfLengthBetween(10, 18);
+        final JwtDateClaimValidator validator = new JwtDateClaimValidator(
+            clock,
+            claimName,
+            TimeValue.ZERO,
+            randomFrom(JwtDateClaimValidator.Relationship.values()),
+            randomBoolean()
+        );
+
+        final SignedJWT jwt = prepareJwt(Map.of(claimName, randomAlphaOfLengthBetween(3, 8)));
+        final ElasticsearchSecurityException e = expectThrows(ElasticsearchSecurityException.class, () -> validator.validate(jwt));
+        assertThat(e.getMessage(), containsString("cannot parse date claim"));
+        assertThat(e.getCause(), instanceOf(ParseException.class));
+    }
+
+    public void testClaimDoesNotExist() throws ParseException {
+        final String claimName = randomFrom(randomAlphaOfLengthBetween(3, 8), "iat", "nbf", "auth_time");
+        final boolean allowNull = randomBoolean();
+
+        final JwtDateClaimValidator validator = new JwtDateClaimValidator(
+            clock,
+            claimName,
+            TimeValue.ZERO,
+            randomFrom(JwtDateClaimValidator.Relationship.values()),
+            allowNull
+        );
+
+        final SignedJWT jwt = prepareJwt(Map.of());
+        if (allowNull) {
+            validator.validate(jwt);
+        } else {
+            final ElasticsearchSecurityException e = expectThrows(ElasticsearchSecurityException.class, () -> validator.validate(jwt));
+            assertThat(e.getMessage(), containsString("missing date claim"));
+        }
+    }
+
+    public void testBeforeNow() throws ParseException {
+        final String claimName = randomFrom(randomAlphaOfLengthBetween(10, 18), "iat", "nbf", "auth_time");
+        final long allowedSkewInSeconds = randomLongBetween(0, 300);
+        final JwtDateClaimValidator validator = new JwtDateClaimValidator(
+            clock,
+            claimName,
+            TimeValue.timeValueSeconds(allowedSkewInSeconds),
+            JwtDateClaimValidator.Relationship.BEFORE_NOW,
+            false
+        );
+
+        final Instant now = Instant.now();
+        when(clock.instant()).thenReturn(now);
+
+        final Instant before = now.minusSeconds(randomLongBetween(1 - allowedSkewInSeconds, 600));
+        validator.validate(prepareJwt(Map.of(claimName, before.getEpochSecond())));
```

Comment:
Should this have a catch and a failure instead of just throwing?
It's a question of personal preference, I think, but this line is the core part of this section of the test, and it feels like it's clearer to understand with something that says _assert-no-exception-is-thrown_, and any test failures will be easier to diagnose.

Reply:
I have no strong view on this. So I changed it to try-catch and rethrow as AssertionError. It is updated here and similar places for other validator tests.